src/components/WorkoutBlock.tsx

```diff
@@ -1,19 +1,34 @@
-import React, { useState } from 'react';
-import { Text, TextInput, TouchableOpacity } from 'react-native';
+import React, { SetStateAction, useState } from 'react';
+import { TextInput, View } from 'react-native';
 import DoneButton from './DoneButton';
-import Ionicon from '@expo/vector-icons/Ionicons';
-import { EditableWorkout } from '../types/EditableWorkout';
+import { EditableWorkout } from '../types';
+import Block from './Block';
 import { BlockContainer } from './BlockContainer';
+import BlockActionsModal from './BlockActionsModal';
+import { useEditableWorkout } from '../hooks/useEditableWorkout';
 
 interface WorkoutBlockProps {
   editing?: string;
+  setEditing: React.Dispatch<SetStateAction<string | undefined>>;
   workout: EditableWorkout;
   updateName: (payload: EditableWorkout) => Promise<void>;
   onPress: (w: EditableWorkout) => void;
+  deleteWorkout: (id: string) => Promise<void>;
+  duplicateWorkout: (payload: EditableWorkout | undefined) => Promise<void>;
 }
 
-const WorkoutBlock: React.FC<WorkoutBlockProps> = ({ editing, workout, updateName, onPress }) => {
+const WorkoutBlock: React.FC<WorkoutBlockProps> = ({
+  editing,
+  setEditing,
+  workout,
+  updateName,
+  onPress,
+  deleteWorkout,
+  duplicateWorkout,
+}) => {
   const [name, setName] = useState<string>(workout.name || '');
+  const [modalVisible, setModalVisible] = useState(false);
+  const editableWorkout = useEditableWorkout(workout.id, true);
```

Comment:
What this is doing is creating a realtime listener for every workout block that checks for live changes to that workout. This means a lot of realtime listeners if we have a lot of workout blocks. Set the second parameter to `false` to avoid this.

Reply:
But we shouldn't need to use this hook in this component. See later comments for suggested changes